R/package.R

```diff
@@ -0,0 +1,157 @@
+##' Updates a dust model in a package. The user-provided code is
+##' assumed to the in `inst/dust` as a series of C++ files; a file
+##' `inst/dust/model.cpp` will be transformed into a file
+##' `src/model.cpp`.
+##'
+##' If your code provides a class `model` then dust will create C++
+##' functions such as `dust_model_alloc` - if your code also includes
+##' names such as this, compilation will fail due to duplicate
+##' symbols.
+##'
+##' We add "Rcpp attributes" to the created functions, and will run
+##' \code{\link{compileAttributes}} on them once the generated code
+##' has been created.
+##'
+##' @param path Path to the package
+##'
+##' @title Create dust model in package
+##' @return Nothing, this function is called for its side effects
+##' @export
+##' @author Rich Fitzjohn
+dust_package <- function(path) {
+  ## 1. check that the package is legit
+  root <- package_validate(path)
+  path_dust <- file.path(root, "inst/dust")
+  path_src <- file.path(root, "src")
+  path_r <- file.path(root, "R")
+
+  ## 2. find target model files
+
+  ## TODO: support alternative location, perhaps not in package - that
+  ## will suit odin.
+  files <- dir(path_dust, pattern = "\\.cpp$")
```

Comment:
We may want to make this more general/user specifiable. `odin.dust` will want to output .cu (and possibly .cpp at the same time) when we're using CUDA

Reply:
I'm in favour of crossing this bridge when we come to it. We'll probably have a bunch of things that come together to tie up a model and associated code